scripts/add.tokens.erc20.mjs

```diff
@@ -0,0 +1,264 @@
+import { load } from 'cheerio';
+import dotenv from 'dotenv';
+import { ethers } from 'ethers';
+import fs from 'fs/promises';
+import fetch from 'node-fetch';
+import { dirname, resolve } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import path from 'path';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+const rootPath = resolve(__dirname, '../');
+
+const envFile = process.env.NODE_ENV === 'production' ? '.env' : '.env.development';
+const envPath = resolve(rootPath, envFile);
+dotenv.config({ path: envPath });
+
+const getArgValue = (argName) => {
+	const argIndex = process.argv.indexOf(argName);
+	return argIndex > -1 && argIndex < process.argv.length - 1 ? process.argv[argIndex + 1] : null;
+};
+
+const ETHERSCAN_API_KEY = getArgValue('--etherscan-api-key') || process.env.VITE_ETHERSCAN_API_KEY;
+
+if (!ETHERSCAN_API_KEY) {
+	console.error(
+		`Missing VITE_ETHERSCAN_API_KEY. Please provide it in ${envFile} or via --etherscan-api-key argument.`
+	);
+	process.exit(1);
+}
+
+const PROD_DASHBOARD_URL = 'https://sv3dd-oaaaa-aaaar-qacoa-cai.raw.icp0.io/dashboard';
+const TESTNET_DASHBOARD_URL = 'https://jzenf-aiaaa-aaaar-qaa7q-cai.raw.icp0.io/dashboard';
+
+const SRC_DIR = 'src/frontend/src/env';
+const SRC_PATH = resolve(rootPath, SRC_DIR);
+
+const fetchHtml = async (url) => {
+	const response = await fetch(url);
+	return await response.text();
+};
+
+const fetchTokenDetails = async (contractAddress) => {
+	const provider = new ethers.providers.EtherscanProvider('homestead', ETHERSCAN_API_KEY);
+	const contract = new ethers.Contract(
+		contractAddress,
+		[
+			'function name() view returns (string)',
+			'function symbol() view returns (string)',
+			'function decimals() view returns (uint8)'
+		],
+		provider
+	);
+
+	const [name, symbol, decimals] = await Promise.all([
+		contract.name(),
+		contract.symbol(),
+		contract.decimals()
+	]).catch((err) => {
+		console.error(`Error fetching token details:\n${err}`);
+		process.exit(1);
+	});
+
+	return { name, symbol, decimals };
+};
+
+const createEnvFile = async (tokenName, tokenDetails, contractAddress, testnetContractAddress) => {
+	const newFileName = `tokens.${tokenName.toLowerCase()}.env.ts`;
+	const newFilePath = path.join(SRC_PATH, newFileName);
+
+	try {
+		await fs.access(newFilePath);
+		console.log(`File ${newFilePath} already exists.`);
+		return;
+	} catch (err) {
+		// File does not exist, proceed with creation
+	}
+
+	const mainnetToken = `${tokenDetails.symbol}_TOKEN`;
+	const testnetToken = `SEPOLIA_${tokenDetails.symbol}_TOKEN`;
+
+	const newContent = `import { ETHEREUM_NETWORK, SEPOLIA_NETWORK } from '$env/networks.env';
```

Comment:
This seems difficult to maintain. Have you thought about for example reading the `tokens.usdc.env` file and doing search and replace in the content?

Reply:
yes, that was my first iteration, but the specifics of searching what needed to be replaced are as difficult to maintain as creating an entire new file, i thought: a lot of specific regex strings that should be re-verified either way in case we change something.

the difference would be that in this case is more clear visually when developing

I am not really sure to be honest, what do you think? which one would be the better way to reduce maintainance?

Reply:
Both seems to have some trade-off in terms of maintenance. Then let's go with your implementation.